### PR TITLE
bug: force ipv4 instead of localhost

### LINKED ISSUE #168: Unable to run on Node.js 17.9.0/18.2.0
If you install node 17.9.0/18.2.0 (using `nvm` or through other means) and run `yarn test` in the `workspaces-js` root directory all tests will fail with the following error:

```
  Rejected promise returned by test. Reason:

  Error (TypedError) {
    context: undefined,
    type: 'RetriesExceeded',
    message: 'Exceeded 10 attempts for http://localhost:1026.',
  }

  › Object.fetchJson (node_modules/near-api-js/lib/utils/web.js:47:15)
  › async node_modules/near-api-js/lib/providers/json-rpc-provider.js:312:34
  › async Object.exponentialBackoff [as default] (node_modules/near-api-js/lib/utils/exponential-backoff.js:7:24)
  › async JsonRpcProvider.sendJsonRpc (node_modules/near-api-js/lib/providers/json-rpc-provider.js:304:26)
  › async JsonRpcProvider.query (node_modules/near-api-js/lib/providers/json-rpc-provider.js:116:22)
  › async Account.findAccessKey (node_modules/near-api-js/lib/account.js:186:31)
  › async Account.signTransaction (node_modules/near-api-js/lib/account.js:91:31)
  › async node_modules/near-api-js/lib/account.js:118:34
  › async Object.exponentialBackoff [as default] (node_modules/near-api-js/lib/utils/exponential-backoff.js:7:24)
  › async Account.signAndSendTransactionV2 (node_modules/near-api-js/lib/account.js:117:24)
```

The last Node.js that works for me is 16.15.0

### PR DESCRIPTION
**Pull request by no2chem**
This PR fixes an issue in Node 18+ which causes tests to never launch.

The root cause is in https://github.com/nodejs/node/issues/40702 - essentially localhost is ambiguously defined as both `127.0.0.1` and `::1`. Node 16 forced `127.0.0.1`, whereas node 18+ just uses whatever the OS gives back as a result of `getaddrinfo(3)` -- on macOS it happens to be `::1` -- but the near sandbox is set to explicitly listen on ipv4 (0.0.0.0:random port).

So, this PR fixes the issue by forcing 127.0.0.1 instead of localhost for the sandbox RPC. 

This PR closes #168

### CODE REVIEW

**Comment by idea404** [comment]
I'd suggest adding these node versions to the CI in `tests-sandbox.yml` to assert this feature

**Comment by no2chem** [comment]
@idea404 done

**Review by volovyks — APPROVED** [review]
We had a similar issue with CLI, thanks @no2chem !
@ailisp do you have access to turn on "On approval" test runs in Workspaces JS? It would be cool to run tests before the merge.

**Comment by ailisp** [comment]
@volovyks I don't have permissions either, let me ask Arthur

**Comment by volovyks** [comment]
@robdefeo for some reason I can not run CI here. Created a separate PR to test Node 18, but faced a dependencie issue that is not fixed here aswell. https://github.com/near/workspaces-js/pull/208

Error:
```
eslint-import-resolver-webpack@0.13.1: The engine "node" is incompatible with this module. Expected version "^16 || ^15 || ^14 || ^13 || ^12 || ^11 || ^10 || ^9 || ^8 || ^7 || ^6". Got "18.7.0"
error Found incompatible module.
```
Do you have the same issue?

**Comment by robdefeo** [comment]
Not sure of the exact cause though I'd suggest upgrading the version of `xo`. In package json is a bit old `"^0.44.0"` where as the current version which appears circa 2 years old. Version `[0.53.1](https://www.npmjs.com/package/xo/v/0.53.1)` is the latest which may have some updates. If this doesn't work then it warrants a further look.

**Comment by volovyks** [comment]
Hey @no2chem ! Looks like we fixed the CI, can you please merge current "main" to your branch?

**Comment by no2chem** [comment]
hi @volovyks - sorry for the delay. rebased and pushed.

File: .github/workflows/tests-sandbox.yml
Comment (gagdiez) [inline .github/workflows/tests-sandbox.yml:12]:
```suggestion
        node-version: [14, 16, 18, 20]
```

**Review by gagdiez — APPROVED** [review]

**Comment by gagdiez** [comment]
all tests are passing, the change is clear, this LGTM @fospring

File: .github/workflows/tests-sandbox.yml
Comment (valeriivanchev) [inline .github/workflows/tests-sandbox.yml:12]: Hey @no2chem! This should resolve the conflicts and we can merge the PR.

```suggestion
        node-version: [16, 18, 20]
```

**Comment by frol** [comment]
@race-of-sloths invite

**Comment by race-of-sloths** [comment]
@frol Thank you for calling!
    
@no2chem Thank you for the contribution! Join [Race of Sloths](https://race-of-sloths.com) by simply mentioning me in your comment/PRs description and start collecting Sloth Points through contributions to open source projects.

<details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/no2chem)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Pull request merged**
